Formula/vulkan-tools.rb

```diff
@@ -0,0 +1,106 @@
+  depends_on "vulkan-loader"
+
+  on_macos do
+    depends_on "molten-vk"
+  end
+
+  on_linux do
+    depends_on "pkg-config" => :build
+    depends_on "libx11"
+    depends_on "libxcb"
+    depends_on "libxkbfile"
+    depends_on "libxrandr"
+    depends_on "wayland"
+    depends_on "wayland-protocols"
+  end
+
+  resource "jsonschema" do
+    url "https://files.pythonhosted.org/packages/36/3d/ca032d5ac064dff543aa13c984737795ac81abc9fb130cd2fcff17cfabc7/jsonschema-4.17.3.tar.gz"
+    sha256 "0f864437ab8b6076ba6707453ef8f98a6a0d512a80e93f8abdb676f737ecb60d"
+  end
+
+  def install
+    venv = virtualenv_create(buildpath/"venv", "python3")
+    resource("jsonschema").stage do
+      system Formula["hatch"].bin/"hatch", "build", "-t", "wheel"
+      venv.pip_install Dir["dist/*.whl"].first
+    end
+
+    if OS.mac?
+      # account for using already-built MoltenVK instead of the source repo
+      inreplace "mac_common.cmake",
+                "${MOLTENVK_DIR}/MoltenVK/icd/MoltenVK_icd.json",
+                "${MOLTENVK_DIR}/share/vulkan/icd.d/MoltenVK_icd.json"
+      Dir["*/macOS/*/*.cmake"].each do |path|
+        inreplace path,
+                  "${MOLTENVK_DIR}/MoltenVK/include",
+                  "${MOLTENVK_DIR}/include"
+        inreplace path,
+                  "${MOLTENVK_DIR}/MoltenVK/dylib/macOS/libMoltenVK.dylib",
+                  "${MOLTENVK_DIR}/lib/libMoltenVK.dylib"
+      end
+    end
+
+    args = [
+      "-DBUILD_ICD=ON",
+      "-DBUILD_CUBE=ON",
+      "-DBUILD_VULKANINFO=ON",
+      "-DINSTALL_ICD=OFF", # we will manually place it in a nonconflicting location
+      "-DGLSLANG_INSTALL_DIR=#{Formula["glslang"].prefix}",
+      "-DVULKAN_HEADERS_INSTALL_DIR=#{Formula["vulkan-headers"].prefix}",
+      "-DVULKAN_LOADER_INSTALL_DIR=#{Formula["vulkan-loader"].prefix}",
+    ]
+    if OS.mac?
+      args += [
+        "-DMOLTENVK_REPO_ROOT=#{Formula["molten-vk"].prefix}",
+      ]
+    end
+    if OS.linux?
+      args += [
+        "-DBUILD_WSI_DIRECTFB_SUPPORT=OFF",
+        "-DBUILD_WSI_WAYLAND_SUPPORT=ON",
+        "-DBUILD_WSI_XCB_SUPPORT=ON",
+        "-DBUILD_WSI_XLIB_SUPPORT=ON",
+      ]
+    end
+    system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    (prefix/"mock_icd").install (buildpath/"build/icd/VkICD_mock_icd.json").realpath,
+                                "build/icd/libVkICD_mock_icd.#{OS.mac? ? "dylib" : "so"}"
+
+    if OS.mac?
+      bin.install_symlink prefix/"vulkaninfo/vulkaninfo"
+      (bin/"vkcube").write_env_script "/usr/bin/open", "-a #{prefix/"cube/vkcube.app"}", {}
+      (bin/"vkcubepp").write_env_script "/usr/bin/open", "-a #{prefix/"cube/vkcubepp.app"}", {}
+    end
+  end
+
+  def caveats
+    <<~EOS
+      The mock ICD files have been installed in
+        #{prefix/"mock_icd"}
+    EOS
+  end
+
+  test do
+    ENV["VK_ICD_FILENAMES"] = prefix/"mock_icd/VkICD_mock_icd.json"
```

Comment:
Do users need to set this environment variable so that the mock ICD files can be found?

Reply:
They do, and it shouldn't be set by default because otherwise it can interfere with any actual Vulkan drivers on the platform.  I'll extend the caveat to cover that.